File: Week_4/OOPS/incpp/PointDemonstration.cpp
#include <iostream>

class Point {
private:
    int x;
    int y;

public:
    Point() {
        x = 0;
        y = 0;
    }

    Point(int x, int y) {
        this->x = x;
        this->y = y;
    }

    void setX(int x) {
        this->x = x;
    }

    void setY(int y) {
        this->y = y;
    }

    void setXY(int x, int y) {
        this->x = x;
        this->y = y;
    }

    void printPoint() {
        std::cout << "Point coordinates: (" << x << ", " << y << ")" << std::endl;
    }
};

int main() {
    Point p1;
    Point p2(10, 20);

    p1.setX(5);
    p1.setY(8);
    p1.printPoint();

    p2.setXY(15, 30);
    p2.printPoint();

    return 0;
}

File: Week_4/OOPS/PointDemonstration.java
class Point {
    private int x;
    private int y;

    public Point() {
        this.x = 0;
        this.y = 0;
    }

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public void setX(int x) {
        this.x = x;
    }

    public void setY(int y) {
        this.y = y;
    }

    public void setXY(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public void printPoint() {
        System.out.println("Point coordinates: (" + x + ", " + y + ")");
    }
}

public class PointDemonstration {
    public static void main(String[] args) {
        Point p1 = new Point();
        Point p2 = new Point(10, 20);

        p1.setX(5);
        p1.setY(8);
        p1.printPoint();

        p2.setXY(15, 30);
        p2.printPoint();
    }
}
